kani-driver/src/cbmc_property_renderer.rs

```diff
@@ -421,6 +437,99 @@ pub fn format_result(
     result_str
 }
 
+pub fn format_result_coverage(properties: &Vec<Property>) -> String {
+    let mut coverage_checks: Vec<CoverageCheck> = Vec::new();
+
+    for prop in properties {
+        let status = &prop.status;
+        let description = &prop.description;
+        let location = &prop.source_location;
+
+        let mut check =
+            CoverageCheck::new("".to_string(), 0, description.to_string(), status.to_string());
+
+        let location_regex = Regex::new(r"(.+):(\d+):").unwrap();
+        let location_msg = format!("{location}\n");
+
+        // Performing the regex match
+        if let Some(captures) = location_regex.captures(&location_msg) {
+            // Accessing the captured groups
+            if let Some(file) = captures.get(1) {
+                let file_name = file.as_str();
+                check.file = file_name.to_string();
+            }
+            if let Some(line_num) = captures.get(2) {
+                let line_number = line_num.as_str().parse::<usize>().unwrap();
+                check.line = line_number;
+            }
+
+            coverage_checks.push(check);
+        } else {
+            println!("No match found.");
+        }
+    }
+
+    post_process_coverage_checks(coverage_checks)
+}
+
+fn post_process_coverage_checks(input: Vec<CoverageCheck>) -> String {
+    let mut formatted_output = String::new();
+
+    let mut sorted_checks: Vec<&CoverageCheck> = input.iter().collect();
+    sorted_checks.sort_by_key(|check| (&check.file, &check.line));
+
+    let filter_checks: Vec<&CoverageCheck> =
+        sorted_checks.iter().filter(|x| x.description == "cover_experiment").cloned().collect();
+
+    let mut files: HashMap<String, Vec<(usize, String)>> = HashMap::new();
+    for check in filter_checks {
+        if !files.contains_key(&check.file) {
+            files.insert(check.file.clone(), vec![(check.line, check.status.clone())]);
+        } else {
+            files.get_mut(&check.file).unwrap().push((check.line, check.status.clone()));
+        }
+    }
+
+    let mut coverage_results: HashMap<String, Vec<(usize, String)>> = HashMap::new();
+    for file in files.keys() {
+        let mut lines: HashSet<usize> = HashSet::new();
+        let mut line_results: Vec<(usize, String)> = Vec::new();
+        for check in files.get(file).unwrap() {
+            lines.insert(check.0);
+        }
+
+        for line in lines.iter() {
+            let satisfiable_statuses: Vec<bool> = files
+                .get(file)
+                .unwrap()
+                .iter()
+                .filter(|(x, _)| *line == *x)
+                .map(|(_, s)| s.contains("SATISFIED"))
+                .collect();
+
+            let covered_status = if satisfiable_statuses.iter().all(|&x| x) {
```

Comment:
You can avoid the `.map(|(_, s)| s.contains("SATISFIED"))` above and do it on `x` here. Let's use better variable names too, please.